File: backend/app/models/movie.py
from sqlalchemy import Column, Integer, String, Float, Date, Text, BigInteger, TIMESTAMP
from sqlalchemy.sql import func
from ..core.database import Base
##from sqlalchemy.orm import relationship
from sqlalchemy.dialects.postgresql import TSVECTOR

class Movie(Base):
    __tablename__ = "movies"

    
    ##genres = relationship("Genre", secondary="movie_genres", viewonly=True)
    ##actors = relationship("MovieActor", back_populates="movie")
    id = Column(Integer, primary_key=True, index=True)
    tmdb_id = Column(Integer, unique=True, nullable=True)
    title = Column(String(200), nullable=False)
    original_title = Column(String(200))
    description = Column(Text)
    release_year = Column(Integer)
    release_date = Column(Date)
    poster_url = Column(String(500))
    backdrop_path = Column(String(500))
    vote_average = Column(Float)
    vote_count = Column(Integer)
    popularity = Column(Float)
    runtime = Column(Integer)
    budget = Column(BigInteger)
    revenue = Column(BigInteger)
    imdb_id = Column(String(20))
    status = Column(String(50))
    tagline = Column(Text)
    tsv = Column(Text)  # tsvector для полнотекстового поиска
    created_at = Column(TIMESTAMP, server_default=func.now())
    tsv = Column(TSVECTOR, nullable=True)  # для полнотекстового поиска

File: flutter/lib/data/models/movie.dart
class Movie {
  final int id;
  final int? tmdbId;
  final String title;
  final String? originalTitle;
  final String? description;
  final int? releaseYear;
  final DateTime? releaseDate;
  final String? posterUrl;          // поле для постера
  final String? backdropPath;       // поле для фонового изображения
  final double? voteAverage;
  final int? voteCount;
  final double? popularity;
  final int? runtime;
  final int? budget;
  final int? revenue;
  final String? imdbId;
  final String? status;
  final String? tagline;
  final String? tsv;
  final DateTime? createdAt;

  Movie({
    required this.id,
    this.tmdbId,
    required this.title,
    this.originalTitle,
    this.description,
    this.releaseYear,
    this.releaseDate,
    this.posterUrl,
    this.backdropPath,
    this.voteAverage,
    this.voteCount,
    this.popularity,
    this.runtime,
    this.budget,
    this.revenue,
    this.imdbId,
    this.status,
    this.tagline,
    this.tsv,
    this.createdAt,
  });

  factory Movie.fromJson(Map<String, dynamic> json) {
    return Movie(
      id: json['id'],
      tmdbId: json['tmdb_id'],
      title: json['title'],
      originalTitle: json['original_title'],
      description: json['description'],
      releaseYear: json['release_year'],
      releaseDate: json['release_date'] != null ? DateTime.parse(json['release_date']) : null,
      posterUrl: json['poster_url'],
      backdropPath: json['backdrop_path'],
      voteAverage: (json['vote_average'] as num?)?.toDouble(),
      voteCount: json['vote_count'],
      popularity: (json['popularity'] as num?)?.toDouble(),
      runtime: json['runtime'],
      budget: json['budget'],
      revenue: json['revenue'],
      imdbId: json['imdb_id'],
      status: json['status'],
      tagline: json['tagline'],
      tsv: json['tsv'],
      createdAt: json['created_at'] != null ? DateTime.parse(json['created_at']) : null,
    );
  }
}